thunder/tests/test_input_as_output_ops.py

```diff
@@ -0,0 +1,56 @@
+from functools import partial
+
+import torch
+from torch.testing import assert_close, make_tensor
+
+import thunder
+import thunder.torch as ttorch
+from thunder.tests.framework import instantiate
+
+import pytest
+
+
+@instantiate(dtypes=(thunder.core.dtypes.inexact,))
+def test_input_as_output_prim(executor, device, dtype):
+    # Tests passing all arguments as function inputs
+    def torch_foo(x, y):
+        z = x * y
+        z = z * z
+        x.copy_(z)
+        # y = y*y
+        return y
```

Comment:
you mean the dce? for a internal decompose like this [func](https://github.com/Lightning-AI/lightning-thunder/blob/1857d1b74e3c98e9c0cf10c6685b0cbd22ce0ee3/thunder/tests/test_input_as_output_ops.py#L23) torch trace is 
```
@torch.no_grad()
@no_autocast()
def computation(x, y):
  # x: "cuda:0 f32[4, 4]"
  # y: "cuda:0 f32[4, 4]"
  z = torch.mul(x, y)  # z: "cuda:0 f32[4, 4]"
    # z = ltorch.mul(x, y)  # z: "cuda:0 f32[4, 4]"
      # z = prims.mul(x, y)  # z: "cuda:0 f32[4, 4]"
  t1 = torch.mul(z, z)  # t1: "cuda:0 f32[4, 4]"
    # t1 = ltorch.mul(z, z)  # t1: "cuda:0 f32[4, 4]"
      # t1 = prims.mul(z, z)  # t1: "cuda:0 f32[4, 4]"
  del z
  copy_(t1, x)
  del t1, x
  return y
```
but nvfuser currently fails because the return statement somehow gets before the other math, I'm still debugging it. When fixed I think we could expect a trace with `fd.add_output(t1, x)` replacing the `copy_(t1, x)`